packages/bitget-wallet/src/lib/bitget-wallet.ts

```diff
@@ -0,0 +1,333 @@
+    //   return [{ accountId, publicKey: undefined }];
+    // }
+
+    // debugger
+    const publicKey = await _state.wallet
+      .getPublicKey
+      // account!.accountId,
+      // options.network.networkId
+      ();
+
+    return [
+      {
+        accountId,
+        publicKey: publicKey ? publicKey.toString() : undefined,
+      },
+    ];
+  };
+
+  const isValidActions = (
+    actions: Array<Action>
+  ): actions is Array<FunctionCallAction> => {
+    return actions.every((x) => x.type === "FunctionCall");
+  };
+
+  const transformActions = (actions: Array<Action>) => {
+    const validActions = isValidActions(actions);
+
+    if (!validActions) {
+      throw new Error(
+        `Only 'FunctionCall' actions types are supported by ${metadata.name}`
+      );
+    }
+
+    return actions.map((x) => x.params);
+  };
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    return transactions.map((transaction) => {
+      return {
+        receiverId: transaction.receiverId,
+        actions: transformActions(transaction.actions),
+        // actions: transaction.actions,
+      };
+    });
+  };
+
+  if (_state.wallet && _state.wallet.isSignedIn()) {
+    setupEvents();
+  }
+
+  return {
+    async signIn({ contractId, methodNames }) {
+      const existingAccounts = await getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      const { accessKey, error } = await _state.wallet.requestSignIn({
+        contractId,
+        methodNames,
+      });
+
+      // debugger;
+      if (!accessKey || error) {
+        await signOut();
+        throw new Error(
+          (typeof error === "string" ? error : error.type) ||
+            "Failed to sign in"
+        );
+      }
+
+      setupEvents();
+
+      return await getAccounts();
+    },
+    signOut,
+    async getAccounts() {
+      return getAccounts();
+    },
+    async verifyOwner({ message }) {
+      logger.log("BitgetWallet:verifyOwner", { message });
+
+      const networkId = options.network.networkId;
+      const accountId = await _state.wallet.getAccountId();
+      const pubKey = await _state.wallet.getPublicKey();
+
+      const block = await provider.block({ finality: "final" });
+
+      const data = {
+        accountId,
+        message,
+        blockId: block.header.hash,
+        // publicKey: Buffer.from(pubKey.data).toString("base64"),
+        publicKey: pubKey,
+        keyType: "0", // TODO: get keyType from sdk wallet
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.signMessage(
+        // new Uint8Array(Buffer.from(encoded)),
+        encoded,
+        accountId,
+        networkId
+      );
+      return {
+        ...data,
+        signature: signed.signature,
+        keyType: signed.keyType,
+      };
+    },
+    // async signMessage(message) {
+    //   debugger
+    // },
```

Comment:
Is this commented function needed here?

Reply:
Thank you very much. We've got rid of that function